### ISSUE TITLE
Distrobox enter hangs indefinitely upon restarting image

### ISSUE THREAD

**Issue by pgosar** [89luca89/distrobox#1382]
Please, before opening a bug:

- make sure you've read the documentation.
- Ensure there isn't already an open issue about this.
- Ensure there isn't already a closed/resolved issue about this.

**Describe the bug**
When using the command distrobox enter <container>, it hangs. I have a decently powerful laptop and let it run for 30 minutes. Upon ending the process, I am met with this message. This happens everytime I restart the container.

`/usr/bin/distrobox-enter: line 680: /home/<user>/.cache/distrobox/.ubuntu-20-04.fifo: No such file or directory`

Distrobox is using docker behind the scenes.

**To Reproduce**
`distrobox create -i ubuntu:20.04`

`distrobox enter ubuntu-20-04` 

**Expected behavior**
I'd expect it to not hang. Note that it does work fine if I end the process and enter the image again. Furthermore I believe there should be a fifo file being created under $home/.cache/distrobox? However no matter what I do such as stopping the running image, remaking it, etc. this file does not show up.

**Logs**
[create.txt](https://github.com/89luca89/distrobox/files/15242262/create.txt) - first command
[enter.txt](https://github.com/89luca89/distrobox/files/15242292/enter.txt) - second command

Attach also the output of `podman logs` or `docker logs`, possibly with `--latest` flag
[docker.txt](https://github.com/89luca89/distrobox/files/15242296/docker.txt) - docker logs

**Desktop (please complete the following information):**
 - Are you using podman, docker or lilipod?  - docker
 - Which version or podman, docker or lilipod? - Docker version 26.1.1, build 4cf5afaefa
 - Which version of distrobox?  - 1.7.2.0
 - Which host distribution? - Arch Linux, kernel 6.8.9
 - How did you install distrobox? - with pacman

**Additional context**

**Comment by pgosar** [comment]
Just docker - tested it with podman 5.0.2

**Comment by rmeno12** [comment]
Am also experiencing the same hanging. If I interrupt with Ctrl-C, I get the same fifo error. Running `distrobox enter <container_name>` again works without hanging or other error messages.
I am on Arch kernel 6.6.30 with distrobox 1.72.1

**Comment by elgq** [comment]
I got this issue too when I switched from podman to docker, and checking the source code of distrobox-enter I noticed the following:

- In line 624 gets the current timestamp in UTC format

```
log_timestamp="$(date -u +%FT%T)"
```

- Then is passed to docker in line 652, to fill a fifo file that is read later for a message.
```
${container_manager} logs --since "${log_timestamp}" -f "${container_name}" \
			> "${app_cache_dir}/.${container_name}.fifo" 2>&1 &
```
- the` --since` parameter as the docker documentation says it expects a timestamp that supposedly is in UTC  as the `docker logs -t <container>` shows.
- but it doesnt show any log line, and the message that is expected to break a loop in lines 674 - 677 doesnt occur. 

```
"container_setup_done"*)
	printf >&2 "\033[32m [ OK ]\n\033[0m"
	kill "${logs_pid}" > /dev/null 2>&1
	break 2	
```
- If the timestamp is changed to localtime in line 624, distrobox-enter works correctly.
```
log_timestamp="$(date +%FT%T)"
```

So or it is a bug with docker --since parameter or the docker/containers need some kind of timezone configuration to match the parameter.

**Comment by kaltec-w** [comment]
I had this same issue on distrobox version 1.7.2.1; modifying the distrobox-enter code to append a 'Z' to the log_timestamps variable does indeed work. I'm not sure why this was marked as completed, since the current distrobox-enter code still does not contain the 'Z'.

**Comment by bfahrenfort** [comment]
Seems to have been fixed in 1.7.2.1-1 (Arch)

**Comment by mounta11n** [comment]
I only have podman and I'm experiencing this issue on openSUSE Leap 15.5,
Kernel 5.14.21,
Podman 4.9.5


However the issue does not occur on openSUSE Tumbleweed,
Kernel 6.11.5-1,
Podman 5.2.5

On Leap my solution is a small workaround for now:

```
#!/bin/bash

# start the process and save its pid
distrobox-enter $1 & PID=$!

# wait to ensure we are not early
sleep 1

# kill the process
kill $PID

echo "..."

# imediately start distrobox-enter process again
distrobox-enter $1
```

I don't know why exactly, but for some reason the whole thing seems to work fine after the container has been killed once.
